connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -256,6 +256,14 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorProps) {
             // do custom connector-specific validation
             Config config = connector.validate(connectorProps);
             ConfigDef configDef = connector.config();
+            if (null == configDef) {
+                throw new BadRequestException(
+                    String.format(
+                        "%s.config() must return a valid ConfigDef.",
```

Comment:
nit: maybe we should mention that the returned config def was null explicitly so the user doesn't have to wonder what was invalid about it?

Reply:
Actually, in addition to clarifying the message, what about also updating the JavaDoc of `Connector.config()` from:

    @return The ConfigDef for this connector.

to

     @return The ConfigDef for this connector; may not be null.